server/controllers/availability.js

```diff
@@ -0,0 +1,100 @@
+const Availability = require("../models/Availability");
+const asyncHandler = require("express-async-handler");
+const mongoose = require("mongoose");
+
+// @route POST /availability
+// @desc create a schedule
+// @access private
+exports.createSchedule = asyncHandler(async (req, res, next) => {
+    const schedule = req.body.schedule
+   
+    let newSchedule = new Availability({
+      profileId: schedule.profileId,
+      schedule: schedule.schedule
+    })
+
+    newSchedule.save((error, schedule) => {
+      if(error){
+        res.status(500)
+        throw new Error(error)
+      } else {
+        res.status(201).send(schedule)
+      }
+    })
+  });
+
+  // @route GET /availability/:scheduleId
+  // @desc get a specific schedule from a profile 
+  // @access public
+  exports.getSchedule = asyncHandler(async (req, res, next) => {
+    const schedule = await Availability.findOne({ _id: req.params.scheduleId})
+
+    if(!schedule){
+      res.status(404)
+      throw new Error("There is no schedule found!")
+    } else {
+      res.status(200).send(schedule)
+    }
+  })
+
+  // @route GET /availability/active
+  // @desc get a profiles active schedule
+  // @access public
+
+  exports.getActiveSchedule = asyncHandler(async (req, res, next) => {
+    const schedules = await Availability.find({ profile: req.query.profileId })
+
+    if (!schedules){
+      res.status(404)
+      throw new Error("There are no schedules!")
+    } else {
+      const activeSchedule = schedules.filter((schedule) => schedule.active )
```

Comment:
We could include this option in the query instead of using the filter. This approach would be faster. `await Availability.find({ profile: req.query.profileId, active: true })`